test/api/test_cpu.cpp

```diff
@@ -22,4 +22,24 @@ TEST_CASE(load_and_run)
     CHECK(bool{shapes_before.front() == outputs.front().get_shape()});
 }
 
+TEST_CASE(zero_parameter)
+{
+    auto p             = migraphx::parse_onnx("constant_fill_test.onnx");
+    auto shapes_before = p.get_output_shapes();
+    p.compile(migraphx::target("cpu"));
+    auto shapes_after = p.get_output_shapes();
+    CHECK(shapes_before.size() == 1);
+    CHECK(shapes_before.size() == shapes_after.size());
+    CHECK(bool{shapes_before.front() == shapes_after.front()});
+    migraphx::program_parameters pp;
+    auto param_shapes = p.get_parameter_shapes();
```

Comment:
if there are no parameters to the program, why are we doing this part of the test?

Reply:
For an onnx file, we may not know whether there are parameters. that is why we need that check.

Reply:
we know this file has no parameters, so I would just add a check for program_parameters.empty() instead.

Reply:
I think if we add a check for param_shapes.empty(), and only if it is false we do the loop, then the following for loop becomes useless.

Reply:
that's fine by me